### ISSUE TITLE
Lando `drush up` not finding updates

### ISSUE THREAD

**Issue by jcmartinez** [lando/lando#1590]
**Tell us about your setup**

I have v3.0.0-rc.14 on Linux Mint Tara.

I have Drupal installed as a multisite.

**Tell us about your `.lando.yml`**

```yml
name: multi7
recipe: drupal7
config:
  webroot: .
  via: nginx
  database: mariadb
proxy:
  mailhog:
    - mail.multi7.lndo.site
  pma:
    - pma.multi7.lndo.site
  appserver_nginx:
    # For custom domains other than lndo.site.
    - multi7.local
    - "*.multi7.local"
    - site1.local
    - site2.local
    - site3.local
services:
  site1db:
    type: mariadb
    creds:
      user: site1_dev
      password: site1_dev
      database: site1_dev
    config:
      database: config/lando/mysql.cnf
  site2db:
    type: mariadb
    creds:
      user: site2_dev
      password: site2_dev
      database: site2_dev
    config:
      database: config/lando/mysql.cnf
  site3db:
    type: mariadb
    creds:
      user: site3_dev
      password: site3_dev
      database: site3_dev
    config:
      database: config/lando/mysql.cnf
  # Spin up a mailhog container.
  mailhog:
    # Use the latest version
    type: mailhog
    # Mailhog expects that these services will be php services.
    hogfrom:
      - appserver
  # PhpMyAdmin
  pma:
    type: phpmyadmin
    hosts:
      - site1db
      - site2db
      - site3db
  appserver:
    build:
      - "[ ! -d /app/drush ] && mkdir -p /app/drush"
      - "[ ! -e /app/drush/drushaliases ] && ln -sf /user/.drush/drushaliases /app/drush/drushaliases"
```

It is also helpful to include other relevant config files. For example, include `pantheon.yml` for a `pantheon` recipe if applicable. Include `php.ini` or `mysql.cnf` if you are overriding our defualt config.

**Tell us about the command you were running**

```bash
$ lando drush @lando.site1 up drupal -vvv
info: Bootstraping...
info: config bootstrap beginning...
debug: Emitting event pre-bootstrap-config
debug: Event pre-bootstrap-config has 0 listeners
verbose: Loading event pre-engine-start priority 2
verbose: Plugin lando-core loaded from /snapshot/lando/build/cli/plugins/lando-core/index.js
debug: Plugin lando-core has {"config":{"appEnv":{"COLUMNS":256,"LANDO":"ON","LANDO_WEBROOT_USER":"www-data","LANDO_WEBROOT_GROUP":"www-data","TERM":"xterm","LANDO_HOST_UID":"1000","LANDO_HOST_GID":"1000","LANDO_HOST_USER":"jc","LANDO_CA_CERT":"/lando/certs/lndo.site.pem","LANDO_CA_KEY":"/lando/certs/lndo.site.key","LANDO_CONFIG_DIR":"/home/jc/.lando","LANDO_DOMAIN":"lndo.site","LANDO_HOST_HOME":"/home/jc","LANDO_HOST_OS":"linux","LANDO_HOST_IP":"192.168.1.201","LANDO_MOUNT":"/app"},"appLabels":{"io.lando.container":"TRUE","io.lando.id":"3bb7738cc79be8a8c03d7e103bedef681c42d508"},"gid":"1000","uid":"1000","username":"jc","caCert":"/home/jc/.lando/certs/lndo.site.pem","caDomain":"lndo.site","caKey":"/home/jc/.lando/certs/lndo.site.key","caProject":"landocasetupkenobi38ahsoka3bb7738cc79be8a8c03d7e103bedef681c42d508"}}
verbose: Plugin lando-events loaded from /snapshot/lando/build/cli/plugins/lando-events/index.js
debug: Plugin lando-events has undefined
verbose: Loading event pre-engine-start priority 1
verbose: Loading event pre-engine-start priority 2
verbose: Plugin lando-networking loaded from /snapshot/lando/build/cli/plugins/lando-networking/index.js
debug: Plugin lando-networking has {"config":{"networkBridge":"lando_bridge_network"}}
verbose: Plugin lando-pantheon loaded from /snapshot/lando/build/cli/plugins/lando-pantheon/index.js
debug: Plugin lando-pantheon has undefined
verbose: Loading event post-bootstrap-config priority 5
verbose: Plugin lando-proxy loaded from /snapshot/lando/build/cli/plugins/lando-proxy/index.js
debug: Plugin lando-proxy has {"config":{"proxy":"ON","proxyCert":"/certs/cert.crt","proxyKey":"/certs/cert.key","proxyName":"landoproxyhyperion5000gandalfedition","proxyDash":"58087","proxyCache":"proxyCache","proxyHttpPort":"80","proxyHttpsPort":"443","proxyHttpFallbacks":["8000","8080","8888","8008"],"proxyHttpsFallbacks":["444","4433","4444","4443"],"proxyDomain":"lndo.site","proxyIp":"127.0.0.1"}}
verbose: Plugin lando-recipes loaded from /snapshot/lando/build/cli/plugins/lando-recipes/index.js
debug: Plugin lando-recipes has undefined
verbose: Plugin lando-services loaded from /snapshot/lando/build/cli/plugins/lando-services/index.js
debug: Plugin lando-services has undefined
verbose: Plugin lando-sharing loaded from /snapshot/lando/build/cli/plugins/lando-sharing/index.js
debug: Plugin lando-sharing has undefined
verbose: Plugin lando-test loaded from /snapshot/lando/build/cli/plugins/lando-test/index.js
debug: Plugin lando-test has {"plugin-test":true}
verbose: Plugin lando-tooling loaded from /snapshot/lando/build/cli/plugins/lando-tooling/index.js
debug: Plugin lando-tooling has undefined
debug: Emitting event post-bootstrap-config
debug: Event post-bootstrap-config has 1 listeners
info: config bootstrap completed.
info: tasks bootstrap beginning...
debug: Emitting event pre-bootstrap-tasks
debug: Event pre-bootstrap-tasks has 0 listeners
debug: Autoloaded task config
debug: Autoloaded task destroy
debug: Autoloaded task info
debug: Autoloaded task list
debug: Autoloaded task logs
debug: Autoloaded task poweroff
debug: Autoloaded task rebuild
debug: Autoloaded task restart
debug: Autoloaded task share
debug: Autoloaded task start
debug: Autoloaded task stop
debug: Autoloaded task version
debug: Autoloaded task init
debug: Autoloaded task ssh
debug: Emitting event post-bootstrap-tasks
debug: Event post-bootstrap-tasks has 0 listeners
info: tasks bootstrap completed.
info: engine bootstrap beginning...
debug: Emitting event pre-bootstrap-engine
debug: Event pre-bootstrap-engine has 0 listeners
debug: Automoved scripts from /snapshot/lando/build/cli/plugins/lando-core/scripts to /home/jc/.lando/scripts and setting 755
debug: Automoved scripts from /snapshot/lando/build/cli/plugins/lando-pantheon/scripts to /home/jc/.lando/scripts and setting 755
debug: Automoved scripts from /snapshot/lando/build/cli/plugins/lando-recipes/scripts to /home/jc/.lando/scripts and setting 755
debug: Automoved scripts from /snapshot/lando/build/cli/plugins/lando-services/scripts to /home/jc/.lando/scripts and setting 755
debug: Automoved scripts from /snapshot/lando/build/cli/plugins/lando-sharing/scripts to /home/jc/.lando/scripts and setting 755
debug: Emitting event post-bootstrap-engine
debug: Event post-bootstrap-engine has 0 listeners
info: engine bootstrap completed.
info: Bootstrap completed.
debug: Emitting event cli-answers
debug: Event cli-answers has 0 listeners
debug: Emitting event cli-drush-answers
debug: Event cli-drush-answers has 0 listeners
debug: Emitting event cli-run
debug: Event cli-run has 0 listeners
debug: Emitting event cli-drush-run
debug: Event cli-drush-run has 0 listeners
debug: Emitting event pre-drush
debug: Event pre-drush has 0 listeners
verbose: Running ["/usr/bin/docker","exec","--tty","--interactive","--user","www-data","--workdir","/app","--env","PHP_MEMORY_LIMIT=-1","multi7_appserver_1","drush","@lando.site1","up","drupal","-vvv"] in /home/jc/websites/multi7 with options {"mode":"attach","detached":false}
debug: With stdio={"stdio":["inherit","inherit","ignore"]}, mode=attach, detached=false, cwd=/home/jc/websites/multi7
Loaded alias @lando.site1 from file /app/drush/drushaliases/lando.aliases.drushrc.php                                                                                                                                                               [notice]
Executing: mysql --defaults-extra-file=/tmp/drush_7pIjj1 --database=site1_dev --host=site1db --silent  < /tmp/drush_ni090n
Executing: mysql --defaults-extra-file=/tmp/drush_bgM1Zh --database=site1_dev --host=site1db --silent  < /tmp/drush_p87DEE
Loading release_info engine.                                                                                                                                                                                                                            [notice]
Loading version_control engine.                                                                                                                                                                                                                         [notice]
Loading package_handler engine.                                                                                                                                                                                                                         [notice]
Executing: wget --version
Loading update_status engine.                                                                                                                                                                                                                           [notice]
env COLUMNS=256  /usr/local/bin/drush  --backend=2 --verbose --root=/app --uri=site1.local  pm-updatestatus drupal 2>&1                                                                                                                             [notice]
Update information last refreshed: Thu, 04/18/2019 - 11:36
 Name    Installed Version  Proposed version  Message    
 Drupal  7.65               7.65              Up to date 


No code updates available.                                                                                                                                                                                                                           [ok]
Command dispatch complete                                                                                                                                                                                                                               [notice]
Command dispatch complete                                                                                                                                                                                                                               [notice]
verbose: /usr/bin/docker,exec,--tty,--interactive,--user,www-data,--workdir,/app,--env,PHP_MEMORY_LIMIT=-1,multi7_appserver_1,drush,@lando.site1,up,drupal,-vvv finished with code 0
debug: Emitting event post-drush
debug: Event post-drush has 0 listeners
```

**Tell us about the error you got**

I would expect that Drush would find Drupal updates.

```bash
Update information last refreshed: Thu, 04/18/2019 - 18:36
 Name    Installed Version  Proposed version  Message    
 Drupal  7.65               7.65              Up to date
```

Note that drupal 7.66 was released on 17 April 2019 (the previous day).

I get the same results whether I run `drush up drupal` to update core or `drush up` to update contributed modules. 

I've also tried avoiding drush aliases using `lando drush -l mysiteurl.com up` and I get the same results.

If I run `drush up drupal` on production (a basic LEMP server) I get the expected results. I.e. Drush identifies the updates (so, not a network issue on Drupal.org).

**Comment by philoSurfer** [comment]
do you see updates in the UI version of update screen?   If yes, then you might not be envoking the correct Drupal instance.   You should always run a drush status to confirm you are bootstrapping the correct Drupal ;)

**Comment by jcmartinez** [comment]
Hi @philoSurfer. Thank you for responding.

It's a multisite installation; therefore all instances share the same source code and Drush should see that there is a new version of Drupal regardless of which instance I hit. Having said that, I always check `lando drush status` when in doubt.

Here's the weird thing. I had updated the source code outside lando and today, while responding to your message, I fired up lando and ran the update command again:

```shell
$ lando drush @lando.site1 up drupal
Update information last refreshed: Mon, 04/22/2019 - 11:39
 Name    Installed Version  Proposed version  Message                   
 Drupal  7.66               7.65              SECURITY UPDATE available 
```
This is a non sense because it offered to "update" to a previous version.

Then, following your suggestion I looked into the update page (/admin/reports/updates) and the page took a bit to load. After loading, it showed that everything was up to date.

I went back to the shell and ran again:

```shell
$ lando drush @lando.site1 up drupal
Update information last refreshed: Mon, 04/22/2019 - 11:46

No code updates available. 
```
This looks like a caching issue. At this point I'm not sure whether this is a Drush issue or a lando issue because when I run Drush on the live server that doesn't have lando, it always picks up the right list of updates.

**Comment by mourogers** [comment]
Does running `lando drush rf` before trying to update help the issue?

**Comment by jcmartinez** [comment]
```$ lando drush rf

In Application.php line 268:
                                                                  
  The pm-refresh command was deprecated. It is no longer useful. ```

**Comment by pirog** [comment]
FWIW Lando *does not* have a caching layer and generally "gets out of the way" as far as the underlying application is concerned. Personally, i've never seen this behavior or heard of someone else having it beyond this issue.

More specifically it's not clear to me _how_ Lando would alter the expected behavior here. That said, if you can give reliable steps to replicate we are happy to look into it more!

**Comment by jcmartinez** [comment]
@pirog 

The steps to replicate are:

1- Install a multisite using the lando recipe that I've included above.

2- When there is a Drupal core or a contrib update available, start lando and run `$ lando drush @lando.site1 up drupal -vvv` as explained above.

Sometimes you'll should see that Drush doesn't find the update.

It may be possible that Drush is caching the updates info and it may be possible that `$ lando drush rf` could solve the problem. If that's the case, this may be a Drush issue; not a Lando one. I could be also some network thing.

I just ran the steps and my installation found an update for Elysia cron this time.

Feel free to close this ticket and I'll report back if I see this happening again.
